www/class/centreonLDAP.class.php

```diff
@@ -781,18 +771,46 @@ private function getBindInfo($id)
     private function debug($msg)
     {
         if ($this->debugImport) {
-            error_log("[" . date("d/m/Y H:i") . "]" . $msg . "\n", 3, $this->debugPath . "ldapsearch.log");
+            error_log("[" . date("d/m/Y H:i") . "] " . $msg . "\n", 3, $this->debugPath . "ldapsearch.log");
         }
     }
 
     /**
-     * Set the error hanlder for LDAP
+     * Override the custom errorHandler to avoid false errors in the log,
      *
+     * @param int $errno The error num
+     * @param string $errstr The error message
+     * @param string $errfile The error file
+     * @param int $errline The error line
+     * @return boolean
+     */
+    private function errorLdapHandler($errno, $errstr, $errfile, $errline): bool
+    {
+        if ($errno === 2 && ldap_errno($this->ds) === 4) {
+            /*
+            Silencing : 'size limit exceeded' warnings in the logs
+            As the $searchLimit value needs to be consistent with the ldap server's configuration and
+            as the size limit error thrown is not related with the results.
+                ldap_errno : 4 = LDAP_SIZELIMIT_EXCEEDED
+                $errno     : 2 = PHP_WARNING
+            */
+            $this->debug("LDAP Error : Size limit exceeded error. This error was not added to php log. "
+                . "Kindly, check your LDAP server's configuration and your Centreon's LDAP parameters.");
+            return self::DISABLED;
+        }
+
+        // throwing all errors
+        $this->debug("LDAP Error : " . ldap_error($this->ds));
+        return  false;
+    }
+
+    /**
+     * Set the error handler for LDAP
      * @see errorLdapHandler
      */
-    private function setErrorHandler()
+    private function setErrorHandler(): void
     {
-        set_error_handler('errorLdapHandler');
+        set_error_handler(array('CentreonLDAP', 'errorLdapHandler'));
```

Comment:
Don't call this method with a static call.
```suggestion
        set_error_handler(array($this, 'errorLdapHandler'));
```